vllm/model_executor/guided_decoding/xgrammar_decoding.py

```diff
@@ -52,65 +49,59 @@ def get_local_xgrammar_guided_decoding_logits_processor(
 @dataclass(frozen=True)
 class TokenizerData:
     """Immutable container for cached tokenizer data."""
+    metadata: str
     encoded_vocab: list[str] = field(default_factory=list)
-    stop_token_ids: list[int] | None = None
-    # These fields are mutually exclusive: `backend_str` is used to create a
-    # TokenizeInfo with `TokenizerInfo.from_huggingface` while `vocab_type` is
-    # used within the constructor of TokenizeInfo
-    backend_str: str | None = None
-    vocab_type: xgr.VocabType | None = None
-
-    def __post_init__(self):
-        # Check for mutual exclusive
-        assert not (self.backend_str and self.vocab_type), \
-            "backend_str and vocab_type are mutual exclusive"
 
 
 class TokenizerDataCache:
     """Cache manager for tokenizer data to avoid repeated processing."""
     _cache: dict[int, TokenizerData] = {}
 
     @classmethod
-    def get_tokenizer_data(cls,
-                           tokenizer: PreTrainedTokenizer) -> TokenizerData:
+    def get_tokenizer_data(
+        cls,
+        tokenizer: PreTrainedTokenizer,
+        vocab_size: int,
+    ) -> TokenizerData:
```

Comment:
nit: we call `hash(tokenizer)` right before calling this function. It might be worthwhile to pass the tokenizer hash to avoid duplicating work.

Reply:
I can include in this PR if that is more convenient...